### PR TITLE
Edit service access control

### LINKED ISSUE #17: UI component to edit service access control (based on wallet addresses)
features: 3, 75

### PR DESCRIPTION
**Pull request by FilipMasar**
Fixes #17 

## What was done?
- add new allow and deny fields (for credentials) when editing or adding a new service
- oceanprotocol/lib version update - https://github.com/oceanprotocol/ocean.js/pull/1855

ocean-node support was done here https://github.com/OceanProtocolEnterprise/ocean-node/pull/12 and here https://github.com/oceanprotocol/ocean-node/pull/698

### CODE REVIEW

File: src/components/Asset/Edit/AddService.tsx [hunk 1]
```diff
@@ -160,6 +163,12 @@ export default function AddService({
         newFiles = filesEncrypted
       }
 
+      const credentials = generateCredentials(
+        undefined,
```
Comment (valentinbica) [inline src/components/Asset/Edit/AddService.tsx:167]: Usually optional parameters should be last

Comment (FilipMasar) [reply src/components/Asset/Edit/AddService.tsx:167]: I am thinking what to do about it, but perhaps it's best to just keep it like that. It is used like that even in the code that I haven't changed e.g. here https://github.com/OceanProtocolEnterprise/market/blob/main/src/components/Publish/_utils.ts#L242
LMK what you think

File: src/components/Asset/Edit/AddService.tsx [hunk 2]
```diff
@@ -169,6 +178,7 @@ export default function AddService({
         datatokenAddress,
         serviceEndpoint: values.providerUrl.url,
         timeout: mapTimeoutStringToSeconds(values.timeout),
+        credentials,
         ...(values.access === 'compute' && {
```
Comment (AdriGeorge) [inline src/components/Asset/Edit/AddService.tsx:182]: should we use enums, to not have string encoded?

File: src/components/Asset/Edit/DebugEditService.tsx [hunk 3]
```diff
@@ -53,13 +56,20 @@ export default function DebugEditService({
...
       const updatedService: Service = {
         ...service,
         name: values.name,
         description: values.description,
         type: values.access,
         timeout: mapTimeoutStringToSeconds(values.timeout),
-        files: updatedFiles, // TODO: check if this works
+        files: updatedFiles, // TODO: check if this works,
+        credentials,
         ...(values.access === 'compute' && {
```
Comment (AdriGeorge) [inline src/components/Asset/Edit/DebugEditService.tsx:73]: same here. enums?

File: src/components/Asset/Edit/EditService.tsx [hunk 4]
```diff
@@ -103,12 +106,19 @@ export default function EditService({
...
+
       const updatedService: Service = {
         ...service,
         name: values.name,
         description: values.description,
         timeout: mapTimeoutStringToSeconds(values.timeout),
-        files: updatedFiles, // TODO: check if this works
+        files: updatedFiles, // TODO: check if this works,
+        credentials: updatedCredentials,
         ...(values.access === 'compute' && {
```
Comment (AdriGeorge) [inline src/components/Asset/Edit/EditService.tsx:122]: same

File: src/components/Asset/Edit/_constants.ts [hunk 5]
```diff
@@ -105,6 +107,14 @@ export const getServiceInitialValues = (
     timeout: secondsToString(service.timeout),
     usesConsumerParameters: service.consumerParameters?.length > 0,
     consumerParameters: parseConsumerParameters(service.consumerParameters),
+    allow:
+      service.credentials?.allow?.find(
+        (credential) => credential.type === 'address'
```
Comment (AdriGeorge) [inline src/components/Asset/Edit/_constants.ts:112]: maybe enum?

**Review by AdriGeorge — APPROVED** [review]
lgtm

**Comment by FilipMasar** [comment]
@AdriGeorge its a good idea to use enums as you pointed out in your comments. But I think it is out of scope of this PR. If I start changing it, it would effect many other files.

**Pull request merged**